engine/api/src/main/java/io/deephaven/engine/table/iterators/CharacterColumnIterator.java

```diff
@@ -4,37 +4,23 @@
 package io.deephaven.engine.table.iterators;
 
 import io.deephaven.engine.primitive.function.CharToIntFunction;
-import io.deephaven.engine.primitive.iterator.CloseablePrimitiveIteratorOfChar;
-import io.deephaven.util.QueryConstants;
 import io.deephaven.util.annotations.FinalDefault;
-import io.deephaven.util.type.TypeUtils;
+import io.deephaven.vector.CharVector;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.PrimitiveIterator;
 import java.util.Spliterator;
 import java.util.Spliterators;
-import java.util.function.Consumer;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
 /**
  * {@link ColumnIterator} implementation for columns of primitive chars.
  */
-public interface CharacterColumnIterator extends ColumnIterator<Character>, CloseablePrimitiveIteratorOfChar {
-
-    @Override
-    @FinalDefault
-    default Character next() {
-        return TypeUtils.box(nextChar());
-    }
-
-    @Override
-    @FinalDefault
-    default void forEachRemaining(@NotNull final Consumer<? super Character> action) {
-        forEachRemaining((final char element) -> action.accept(TypeUtils.box(element)));
-    }
+public interface CharacterColumnIterator extends ColumnIterator<Character>, CharVector.Iterator {
```

Comment:
It feels sort of offensive that this is a `CharVector.Iterator`, but it doesn't seem to have any real correctness issues. Maybe we could/should pull those out as top-level `DeephavenValueIterator` classes or something, but it doesn't make logical sense to me that an `XYZColumnIterator` is an `XYZVector.Iterator`.

The overrides you've kept in these classes are probably better off in the superinterfaces, though, if you have a way to implement `remaining()`.

The overrides for the no-argument versions of the `stream()` methods (primitive and boxed) have lost their internal sizing with the loss of `remaining()`.